apps/core/app/[locale]/(default)/cart/_components/coupon-code.tsx

```diff
@@ -0,0 +1,138 @@
+'use client';
+
+import { Button } from '@bigcommerce/components/button';
+import { Field, FieldControl, FieldMessage, Form, FormSubmit } from '@bigcommerce/components/form';
+import { Input } from '@bigcommerce/components/input';
+import { AlertCircle, Loader2 as Spinner } from 'lucide-react';
+import { useTranslations } from 'next-intl';
+import { useEffect, useState } from 'react';
+import { useFormStatus } from 'react-dom';
+import { toast } from 'react-hot-toast';
+
+import { getCheckout } from '~/client/queries/get-checkout';
+import { ExistingResultType } from '~/client/util';
+
+import { applyCouponCode } from '../_actions/apply-coupon-code';
+import { removeCouponCode } from '../_actions/remove-coupon-code';
+
+type Checkout = ExistingResultType<typeof getCheckout>;
+
+const SubmitButton = () => {
+  const t = useTranslations('Cart.SubmitCouponCode');
+  const { pending } = useFormStatus();
+
+  return (
+    <Button className="items-center px-8 py-2" disabled={pending} variant="secondary">
+      {pending ? (
+        <>
+          <Spinner aria-hidden="true" className="animate-spin" />
+          <span className="sr-only">{t('spinnerText')}</span>
+        </>
+      ) : (
+        <span>{t('submitText')}</span>
+      )}
+    </Button>
+  );
+};
+
+export const CouponCode = ({ checkout }: { checkout: ExistingResultType<typeof getCheckout> }) => {
+  const t = useTranslations('Cart.CheckoutSummary');
+  const [showAddCoupon, setShowAddCoupon] = useState(false);
+  const [selectedCoupon, setSelectedCoupon] = useState<Checkout['coupons'][number] | null>(
+    checkout.coupons[0] || null,
```

Comment:
Can you explain the benefit of this?  📝

Reply:
It's mostly beneficial for retrieving the last item in an array, as you can do `arr.at(-1)` instead of `arr[arr.length - 1]`. Again, it's a 🍹, mostly just trying to push towards accessing array indexes via that method.